File: tests/machine/x/fortran/two-sum.f90
program two_sum
  implicit none
  integer :: result
    integer :: n
    integer :: i
      integer :: j
  result = twoSum((/2,7,11,15/),9)
  print *, result(((0)+1))
  print *, result(((1)+1))
  contains
  recursive integer function twoSum(nums,target) result(res)
    integer, intent(in) :: nums
    integer, intent(in) :: target
    n = size(nums)
    do i = 0, n
      do j = (i + 1), n
        if (((nums(((i)+1)) + nums(((j)+1))) == target)) then
          res = (/i,j/)
          return
        end if
      end do
    end do
    res = (/-1,-1/)
    return
  end function twoSum
end program two_sum

File: tests/machine/x/java/two-sum.java
import java.util.*;

public class TwoSum {
    static List<Integer> twoSum(List<Integer> nums, int target) {
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (Objects.equals(((Number)nums.get(i)).doubleValue() + ((Number)nums.get(j)).doubleValue(), target)) {
                    return Arrays.asList(i, j);
                }
            }
        }
        return Arrays.asList(-1, -1);
    }
    public static void main(String[] args) {
    List<Integer> result = twoSum(Arrays.asList(2, 7, 11, 15), 9);
    System.out.println(result.get(0));
    System.out.println(result.get(1));
    }
}
